### PR TITLE
fix(RHINENG-13652): Add system profile field to HostOut

### PR DESCRIPTION
**Pull request by jpramos123**
# Overview

This PR is being created to address [RHINENG-13652](https://issues.redhat.com/browse/RHINENG-13652).

It adds `system_profile` field to `HostOut` object in the OpenAPI spec

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -1753,6 +1753,12 @@ components:
               type: array
               items:
                 $ref: '#/components/schemas/GroupOut'
+            system_profile:
```
[the remark this answers is not in the document]
Comment (kruai) [reply swagger/api.spec.yaml:1756]: This is just adding it to the schema as an optional field. This _allows_ system profile data to be returned on the host object, but does not require it or change any of the output.

**Review by kruai — CHANGES_REQUESTED** [review]

File: swagger/api.spec.yaml [hunk 2]
```diff
@@ -1753,6 +1753,12 @@ components:
...
+              description: >-
+                The host system profile object
+              type: object
+              items:
+                $ref: '#/components/schemas/SystemProfileByHostOut'
```
[the remark this answers is not in the document]
Comment (kruai) [reply swagger/api.spec.yaml:1761]: You are likely looking for `#/components/schemas/SystemProfile`

[the remark this answers is not in the document]
Comment (jpramos123) [reply swagger/api.spec.yaml]: Done!

File: swagger/api.spec.yaml [hunk 3]
```diff
@@ -1753,6 +1753,12 @@ components:
               type: array
               items:
                 $ref: '#/components/schemas/GroupOut'
+            system_profile:
+              description: >-
+                The host system profile object
+              type: object
+              items:
+                $ref: '#/components/schemas/SystemProfile'
```
Comment (kruai) [inline swagger/api.spec.yaml:1761]: The `items` field is used for arrays, not objects. If you run the API service locally and look at the swagger docs, you can see that this change is not being reflected in the example response or response schema.
```suggestion
            system_profile:
              $ref: '#/components/schemas/SystemProfile'
```

**Review by kruai — CHANGES_REQUESTED** [review]

File: swagger/api.spec.yaml [hunk 4]
```diff
@@ -1753,6 +1753,11 @@ components:
               type: array
               items:
                 $ref: '#/components/schemas/GroupOut'
+            system_profile:
+              description: >-
+                The host system profile object
+              type: object
+              $ref: '#/components/schemas/SystemProfile'
```
Comment (kruai) [inline swagger/api.spec.yaml:1760]:
```suggestion
            system_profile:
              $ref: '#/components/schemas/SystemProfile'
```

**Review by kruai — CHANGES_REQUESTED** [review]
Didn't quite make the requested change; I've re-suggested the change

**Review by kruai — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]

**Pull request merged**